src/main/java/org/sefglobal/scholarx/service/MentorService.java

```diff
@@ -66,11 +68,15 @@ public Mentor getMentorById(long id) throws ResourceNotFoundException {
      * @param id             which is the {@link Mentor} to be updated
      * @param enrolmentState which is the {@link EnrolmentState} of the program to be updated
      * @return the updated {@link Mentor}
-     *
      * @throws ResourceNotFoundException is thrown if the requesting {@link Mentor} doesn't exist
+     * @throws BadRequestException is thrown if the requesting {@link EnrolmentState} is not eligible for a mentor
      */
     public Mentor updateState(long id, EnrolmentState enrolmentState)
-            throws ResourceNotFoundException {
+            throws ResourceNotFoundException, BadRequestException {
+        if (!eligibleEnrolmentStatesForMentors.contains(enrolmentState)) {
```

Comment:
```suggestion
        final List<EnrolmentState> mentorStates = ImmutableList.of(EnrolmentState.APPROVED, EnrolmentState.REJECTED);
        if (!mentorStates.contains(enrolmentState)) {
```
Let's declare the mentorStates List in here itself, because there are no other usages of that.

Reply:
And before throwing the Exception, let's create a String with this error message. 

Suggestion:
```
            String msg = "Error, Mentor with id: " + id + " cannot be updated. " +
                         enrolmentState + " is not an applicable state.";
            log.error(msg);
            throw new BadRequestException();
```

Reply:
Done the part with the error message.
I would like to argue against creating the list inside the method simply because it will be created each time the method is called...putting pressure on the GC.
Another reason for having the list as a field is that probably in the future that list can be provided from outside: created based on a config property etc.

Reply:
@Piumal1999 What do you think?

Reply:
Awesome @posabogdanpetre . Can you add this one also

> Let's declare the mentorStates List in here itself, because there are no other usages of that.

```
        final List<EnrolmentState> validMentorStates = ImmutableList.of(EnrolmentState.APPROVED, EnrolmentState.REJECTED);
        if (!validMentorStates.contains(enrolmentState)) {
```

Reply:
@Piumal1999 Added a comment above related to this...if the comment is not convincing you then I can do this change.

Reply:
Oh Sorry, I didn't see your message. 

> I would like to argue against creating the list inside the method simply because it will be created each time the method is called...putting pressure on the GC.

Yeah, that's right. Your method is better. Thank you